src/shapes/textbox.class.js

```diff
@@ -78,6 +86,33 @@
      */
     splitByGrapheme: false,
 
+    /**
+     * While editing handle differently
+     * @private
+     * @param {string} key
+     * @param {*} value
+     */
+    _set: function (key, value) {
+      if (key === 'width') {
+        if (typeof this.maxWidth === 'number' && this.maxWidth < value) {
+          value = this.maxWidth;
+        }
+        if (typeof this.minWidth === 'number' && this.minWidth > value) {
+          value = this.minWidth;
+        }
+      }
+      if (key === 'maxWidth' && this.minWidth > value) {
+        value = this.minWidth;
+      }
+      this.callSuper('_set', key, value);
+      if (key === 'minWidth' && this.maxWidth < value) {
+        this._set('maxWidth', value);
+      }
+      if ((key === 'maxWidth' && this.width > value) || (key === 'minWidth' && this.width < value)) {
+        this._set('width', value);
+      }
```

Comment:
mmmm not my favorite side effect.
This a new behavior, makes it breaking ( fine ) and should be written on top of minWidth and maxWidth.
Warning, setting this will also immediately affect width if condition apply.
Consider that setting this without re-rendering could be a recipe for disaster.
A valid alternative would be the opposite, if you are setting maxWidth or minWidth out of range, console warn and do nothing.
I think i prefer the do nothing option.

Reply:
I agree. It's better to let the dev set width as well.